File: chain-of-responsibility/java/JPEGHandler.java
package ChainOfResponsibility;

// JPEG dosyalarını dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
class JPEGHandler extends BaseHandler {

    @Override
    public void handleRequest(Image image) {
        if (image.extension == "JPEG") {
            System.out.println("JPEG to PNG");
            // JPEG işlemine ait dönüştürme kodları işlemleri.
        } else {
            // Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
            nextHandler.handleRequest(image);
        }
    }
}

File: chain-of-responsibility/typescript/jpeg-handler.ts
import {BaseHandler} from "./base-handler";
import {Image} from "./image";

// JPEG dosyalarını dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
export class JPEGHandler extends BaseHandler {
  public handleRequest(image: Image): void {
    if(image.extension === 'JPEG') {
      console.log('JPEG to PNG');
      // JPEG işlemine ait dönüştürme kodları işlemleri.
    } else {
      // Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
      this.nextHandler.handleRequest(image);
    }
  }
}
